internal/resources/kubeadm_phases.go

```diff
@@ -159,6 +160,12 @@ func (r *KubeadmPhase) GetKubeadmFunction(ctx context.Context, tcp *kamajiv1alph
 		}, nil
 	case PhaseClusterAdminRBAC:
 		return func(c clientset.Interface, configuration *kubeadm.Configuration) ([]byte, error) {
+			// If bootstrap RBAC is configured, use that instead of default kubeadm behavior
+			if tcp.Spec.Bootstrap != nil && tcp.Spec.Bootstrap.RBAC != nil && tcp.Spec.Bootstrap.RBAC.Enabled {
+				return r.createBootstrapRBAC(ctx, c, tcp)
+			}
+
+			// Fallback to original kubeadm behavior for backward compatibility
```

Comment:
I like the idea of making toggleable the RBAC setup, but I would avoid having a separate function such as `createBootstrapRBAC`.

We should make `Bootstrap` and `RBAC` enabled by default, with the same defaults as `Kubernetes`, and try to resuse the underlying `kubeadm` library for DRY purposes/duplication of code.